internal/server/handlers/rbac/backup_storage_test.go

```diff
@@ -0,0 +1,158 @@
+package rbac
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	everestv1alpha1 "github.com/percona/everest-operator/api/v1alpha1"
+	"github.com/percona/everest/internal/server/handlers"
+	"github.com/percona/everest/pkg/rbac"
+)
+
+func TestRBAC_BackupStorage(t *testing.T) {
+	t.Parallel()
+
+	t.Run("ListBackupStorages", func(t *testing.T) {
+		t.Parallel()
+		next := handlers.MockHandler{}
+		h := &rbacHandler{
+			next: &next,
+			log:  zap.NewNop().Sugar(),
+		}
+
+		next.On("ListBackupStorages",
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+		).Return(
+			&everestv1alpha1.BackupStorageList{
+				Items: []everestv1alpha1.BackupStorage{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "backup-storage-1",
+							Namespace: "default",
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "backup-storage-2",
+							Namespace: "default",
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "backup-storage-3",
+							Namespace: "default",
+						},
+					},
+				},
+			},
+			nil,
+		)
+
+		testCases := []struct {
+			policy string
+			outLen int
+		}{
+			{
+				policy: newPolicy(
+					"p, role:test, backup-storages, read, default/backup-storage-1",
+					"g, test-user, role:test",
+				),
+				outLen: 1,
+			},
+			{
+				policy: newPolicy(
+					"p, role:test, backup-storages, read, default/*",
+					"g, test-user, role:test",
+				),
+				outLen: 1,
```

Comment:
Shouldn't this be 3?

Reply:
Yes, you're right, I didn't pay a lot of attention to the tests here because it was just a rough POC 🙈 I've fixed it now ...